src/ObjectUtils/LinuxMap.cpp

```diff
@@ -155,74 +194,64 @@ class FileMappedIntoMemory {
         absl::StrFormat("No, anonymous executable map at %#x-%#x does NOT belong to \"%s\"",
                         map_start, map_end, file_path_);
 
-    if (cached_coff_file_ == nullptr) {
-      // Don't even try to create an ObjectFile from character or block devices.
-      if (absl::StartsWith(file_path_, "/dev/")) {
-        ORBIT_LOG("%s: this is a device file", error_message);
-        coff_text_section_map_might_be_encountered_ = false;
-        return false;
-      }
-
-      auto object_file_or_error = CreateObjectFile(file_path_);
-      if (object_file_or_error.has_error()) {
-        ORBIT_LOG("%s: %s", error_message, object_file_or_error.error().message());
-        coff_text_section_map_might_be_encountered_ = false;
-        return false;
-      }
-
-      // Remember: we are only detecting anonymous maps that correspond to .text sections of PEs,
-      // because loadable sections of ELF files can always be file-mapped.
-      if (!object_file_or_error.value()->IsCoff()) {
-        ORBIT_LOG("%s: object file is not a PE", error_message);
-        coff_text_section_map_might_be_encountered_ = false;
-        return false;
-      }
-
-      cached_coff_file_ = std::move(object_file_or_error.value());
+    // Remember: we are only detecting anonymous maps that correspond to executable sections of PEs,
+    // because loadable segments of ELF files can always be file-mapped.
+    if (!object_file_->IsCoff()) {
+      ORBIT_LOG("%s: object file is not a PE", error_message);
+      return;
     }
 
-    ORBIT_CHECK(cached_coff_file_ != nullptr);
+    if (first_map_start_ < first_map_offset_) {
+      // In this case, base_address would be the result of an underflow. This shouldn't happen.
+      ORBIT_ERROR("%s: the memory address of the first byte of the PE is negative", error_message);
+      return;
+    }
+    // The start address of the map in which the first byte of the PE is (or would be) mapped. It is
+    // page-aligned because it is derived from a file mapping.
+    const uint64_t base_address = first_map_start_ - first_map_offset_;
```

Comment:
I realized that subtracting the offset here is wrong, because it doesn't necessarily equal the RVA of the section. I will probably just require that the first offset be zero. This is the expected case, as the first map should correspond to the headers and they are at the beginning of the file.

Reply:
Changed.